### PR TITLE
feat:(Add submenu option)

### PR DESCRIPTION
**Pull request by danlaush**
~Note: This is a work in progress—PR created for discussion.~ PR ready for review now.

The goal of this work is to allow public-navigation integrations to include a 2nd-level menu, such as appears on the [IBAN checker](https://transferwise.com/gb/iban/) and [blog](https://transferwise.com/gb/blog/). 

Example:
Screenshot 2019-09-26 at 12 43 29

There's a couple things I think would be good to discuss/get agreement on before I go much further on this.

- Approach, 2 ideas:
    - Allow the consuming application to pass in the items to be shown in the second-level nav (see Localisation) 
    - Add the menu structure for each project using the public-navigation and allow the consuming application to pass in a key that corresponds to the correct submenu items.
- Localisation
    - @brettanomyces suggested that it might be easier/more maintainable for integrators to, instead of adding translation keys to public-navigation for each project's use case, pass in links which have already been localised in the consuming application. This allows public-navigation to be agnostic and doesn't require an update when the application requires a change in the menu, which would (presumably) be in a different team.
- Styles
    - I've made a stab at using the Bootstrap submenu classes to add the second-level nav, see below. This is how the existing IBAN checker et al. implement this. There's a few little adjustments that I think will need to be made, as there's some things in there like `class="navbar--inverse navbar-inverse"` (the former from the react component, the latter from bootstrap) and some other little things caused by overlapping CSS.
    - It might be easiest to work through these on a case-by-case basis, but I'm still figuring out so it would be great to chat through it with someone who has more history on it.

Current effort (26 Sep):
Screenshot 2019-09-25 at 15 51 21

### CODE REVIEW

File: docs/Docs.less [hunk 1]
```diff
@@ -8,7 +8,7 @@
   left: 0;
   right: 0;
 
-  height: 56px;
+  height: 64px;
```
Comment (danlaush) [inline docs/Docs.less]: Per designs, when the nav has a submenu it needs to be taller. Can be handled in JS logic.

**Comment by mikknarusk** [comment]
@danlaush I think this approach makes sense. Let's allow applications to pass in a submenu array, with strings already translated. I assume we should still pass in unlocalised urls (eg `/:locale/borderless` as it doesn't require any configuration on the public-nav side and works out of the box. Also keeps the logic of mapping locales to urls in a single place.

Nice job, looks easy enough, although mobile could require more fine tuning :)

**Comment by danlaush** [comment]
@mikknarusk This is now ready for review! The git history got a bit cooked when I rebased, unfortunately. 🤔 Hate keeping branches open for so long.

Screenshot 2019-10-31 at 15 08 35

File: src/PublicNavigation/Navigation/Navigation.js [hunk 2]
```diff
@@ -37,9 +38,27 @@ class Navigation extends Component {
...
+           * were not required to render the top-level nav, but they are necessary
+           * for the subnav to render correctly. In order to limit exposure to more
+           * CSS than is required, the Bootstrap inverse class is only added
+           * when a subnav is being used.
+           */
+          subnavItems.length ? 'subnav navbar-inverse' : '',
+          className,
+        ]
+          .filter(i => i) // remove falseys
+          .join(' ')}
```
Comment (fergusjordantw) [inline src/PublicNavigation/Navigation/Navigation.js]: Can we just use `classNames` here instead as it's already a project dependency?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
+# v3.4.5
+## Added support for subnav
```
Comment (fergusjordantw) [inline CHANGELOG.md]: Should this be a minor version bump?

**Comment by danlaush** [comment]
@mikknarusk @fergusjordantw I'm closing this PR in favour of https://github.com/transferwise/public-navigation/pull/121 where I cleaned up the commit history. Fergus, I've addressed all your comments here in that branch.

**Pull request closed without merging**